packages/flutter_tools/lib/src/flutter_manifest.dart

```diff
@@ -108,10 +108,9 @@ class FlutterManifest {
   /// Can be null if version isn't set or has a wrong format.
   String get buildNumber {
     if (appVersion != null && appVersion.contains('+')) {
-      final String value = appVersion.split('+')?.elementAt(1);
-      return value;
+      return appVersion.split('+')?.elementAt(1);
```

Comment:
```suggestion
      return appVersion.split('+')[1];
```
as far as I can tell, split can't return null, so there's no reason to check for it.